src/coreclr/src/vm/comdelegate.cpp

```diff
@@ -1146,6 +1146,9 @@ PCODE COMDelegate::ConvertToCallback(MethodDesc* pMD)
     }
     CONTRACTL_END;
 
+    // Check the method for proper usage.
+    ThrowIfInvalidUnmanagedCallersOnlyUsage(pMD);
```

Comment:
I think this check would be better in `UMThunkMarshInfo::RunTimeInit` that executes when the method is called for the first time - to match the behavior for non-x86.

Reply:
See comments about this function above. I chose this function because it is the choke point for the `UnmanagedCallersOnly` scenario now.

Reply:
>to match the behavior for non-x86.

I see the argument here, I will do that. Thoughts on the renaming this function since it really is only for `UnmanagedCallersOnly` scenario?